G/gb/build_tarballs.jl

```diff
@@ -0,0 +1,51 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "gb"
+version = v"0.17.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/ederc/gb.git", "ec98ed8deeeb47e3e7bdf38eb947453bfa6b23b5")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd gb/
+./autogen.sh 
+./configure --enable-shared --disable-static --prefix=${prefix} --build=${MACHTYPE} --host=${target} --with-gmp=${prefix}
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "macos"; ),
+    Platform("x86_64", "freebsd"; )
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libgb--no-undefined", :libgb)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"))
```

Comment:
```suggestion
    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"), v"6.1.2"),
    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
```

Reply:
Can I ask why?

Reply:
The reason is rather complex and has to do with the fact that `GMP` ships with Julia; on Julia v1.5 we ship `GMP 6.1.2` and on Julia 1.6 we (will) ship `GMP 6.2.0`.  GMP is backwards compatible, but if you compile against `GMP 6.2.0` (which would be the default choice of the build algorithm) you'll only be compatible with GMP 6.2.0+.  So we explicitly build against an older version so that older versions of Julia can use this library.